lib/input.js

```diff
@@ -28,9 +28,9 @@ function _createInputDescriptor (input, inputOptions, containerOptions) {
   if (is.string(input)) {
     // filesystem
     inputDescriptor.file = input;
-  } else if (is.buffer(input)) {
-    // Buffer
-    inputDescriptor.buffer = input;
+  } else if (is.uint8Array(input)) {
+    // Buffer or Uint8Array
+    inputDescriptor.buffer = Buffer.from(input);
```

Comment:
The Node.js docs suggest this will copy data rather than create a reference/view.

https://nodejs.org/dist/latest-v14.x/docs/api/buffer.html#buffer_class_method_buffer_from_buffer

Perhaps use `Buffer.from(input.buffer)` instead?

Perhaps also have this as an additional `else if (is.uint8Array(input))` condition, keeping the original `else if (is.buffer(input))` intact.

Reply:
About the copy/mutable thing:
Yes, that's true, and I have to confess that I don't know a lot about "buffers" generally speaking, because I grew up with Javascript and not with languages where you actually **have** to deal with memory and buffers directly all the time. So the first time I came across the "Arraybuffer is just a reference" and "whatever you do to this buffer, also changes things in every reference buffer" kind of thing, it weirded me out, because most of the times things in Javascript don't behave like this. I get that sharing the same allocated memory is intended behaviour, but I still feel like it confuses more than it makes things faster/better. Look at [this small example](https://runkit.com/nnmrts/buffers).

Yada yada yada, I think for sharp, and especially in this scenario, where sharp later passes that buffer on to libvips and then gives us back a new one, the whole copy/reference stuff is irrelevant. We are copying it/creating a new one anyway later on, afaik.

But, A, of course, you could still argue that using the underlying ArrayBuffer is more explicit and B, I often read that this has a lot to do with performance, so I guess using ArrayBuffer is faster? I don't know. But considering all of that, I'll look into how to use ArrayBuffer correctly here, so we don't make useless copies.

About the condition thing:
Sure, I can do that if you really want, but as I said before, node's Buffer is just a subclass of Uint8Array, so this:

```js
const input = Buffer.from([1,2,3,4]); // can be Buffer or uint8array

let result; // should be Buffer

if (input instanceof Buffer) {
	result = buffer;
}
else if (input instanceof Uint8Array) {
	result = Buffer.from(input.buffer);
}
```

is exactly the same as:

```js
const input = Buffer.from([1,2,3,4]); // can be Buffer or uint8array

let result; // should be Buffer

if (input instanceof Uint8Array) {
	result = Buffer.from(input.buffer);
}
```

Using `else if (is.uint8Array(input))` is probably easier to read/follow though, true.